src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -95,6 +95,244 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
+            bool hasParamChanges = false;
+            if (_parameters.Length > 0)
+            {
+                var paramMergedTypesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(_parameters.Length);
+                var paramRefCustomModifiersBuilder = ArrayBuilder<ImmutableArray<CustomModifier>>.GetInstance(_parameters.Length);
+                for (int i = 0; i < _parameters.Length; i++)
+                {
+                    var thisParam = _parameters[i];
+                    var otherParam = signature._parameters[i];
+                    Debug.Assert(thisParam.RefKind == otherParam.RefKind);
+                    var paramVariance = (variance, thisParam.RefKind) switch
+                    {
+                        (VarianceKind.In, RefKind.None) => VarianceKind.Out,
+                        (VarianceKind.Out, RefKind.None) => VarianceKind.In,
+                        _ => VarianceKind.None,
+                    };
+
+                    var mergedParameterType = thisParam.TypeWithAnnotations.MergeEquivalentTypes(otherParam.TypeWithAnnotations, paramVariance);
+                    paramMergedTypesBuilder.Add(mergedParameterType);
+                    if (!mergedParameterType.IsSameAs(thisParam.TypeWithAnnotations))
+                    {
+                        hasParamChanges = true;
+                    }
+
+                    var mergedCustomModifiers = mergeCustomModifiers(thisParam.RefCustomModifiers, otherParam.RefCustomModifiers, paramVariance);
+                    paramRefCustomModifiersBuilder.Add(mergedCustomModifiers);
+                    if (mergedCustomModifiers != RefCustomModifiers)
+                    {
+                        hasParamChanges = true;
+                    }
+                }
+
+                if (hasParamChanges)
+                {
+                    mergedParameterTypes = paramMergedTypesBuilder.ToImmutableAndFree();
+                    mergedParamRefCustomModifiers = paramRefCustomModifiersBuilder.ToImmutableAndFree();
+                }
+                else
+                {
+                    paramMergedTypesBuilder.Free();
+                    paramRefCustomModifiersBuilder.Free();
+                    mergedParameterTypes = ParameterTypesWithAnnotations;
+                }
+            }
+
+            if (hasParamChanges || !mergedReturnType.IsSameAs(ReturnTypeWithAnnotations) || mergedReturnRefCustomMods != RefCustomModifiers)
+            {
+                return SubstiteParameterSymbols(mergedReturnType, mergedParameterTypes, mergedReturnRefCustomMods, mergedParamRefCustomModifiers);
+            }
+            else
+            {
+                return this;
+            }
+
+            static ImmutableArray<CustomModifier> mergeCustomModifiers(ImmutableArray<CustomModifier> thisRefCustomModifiers, ImmutableArray<CustomModifier> otherRefCustomModifiers, VarianceKind variance)
```

Comment:
>variance [](start = 195, length = 8)

Variance isn't applicable to custom modifiers, they do not part of data flow information

Reply:
Function will be removed.

---
In reply to: [406963755](https://github.com/dotnet/roslyn/pull/43041#discussion_r406963755) [](ancestors = 406963755)